src/test/regress/sql/other_databases.sql

```diff
@@ -0,0 +1,39 @@
+CREATE SCHEMA other_databases;
+SET search_path TO other_databases;
+
+SET citus.next_shard_id TO 10231023;
+
+CREATE DATABASE other_db1;
+
+\c other_db1
+SET citus.is_main_db TO false;
+
+SHOW citus.is_main_db;
+SHOW citus.main_db;
+
+CREATE USER other_db_user1;
```

Comment:
These are good tests assuming all nodes are up, but I think we should have some failure tests too. Where we test the 2PC failure/retry logic.

Reply:
Re-iterating that this PR really needs failure tests to check if the abortion logic is correct. Currently it does not look like it is (see other comment for details).